app/HMS/Repositories/Doctrine/DoctrineRoleRepository.php

```diff
@@ -0,0 +1,48 @@
+<?php
+
+namespace HMS\Repositories\Doctrine;
+
+use HMS\Entities\Role;
+use Doctrine\ORM\EntityRepository;
+use HMS\Repositories\RoleRepository;
+use Doctrine\Common\Collections\ArrayCollection;
+
+class DoctrineRoleRepository extends EntityRepository implements RoleRepository
+{
+    /**
+     * Finds all entities in the repository.
+     *
+     * @return ArrayCollection The entities.
+     */
+    public function findAll()
+    {
+        $entities = parent::findAll();
+        $roles = [];
+        foreach ($entities as $entity) {
+            $roles[$entity->getName()] = $entity;
+        }
+
+        return new ArrayCollection($roles);
```

Comment:
I think the change of array vs ArrayCollection should be a separate PR once we have chosen which we want. And this PR should be mergers as is.